fplus-http-server/src/router/application.rs

```diff
@@ -311,26 +338,36 @@ pub async fn all_applications() -> impl Responder {
 #[get("/application/active")]
 pub async fn active(query: web::Query<GithubQueryParams>) -> impl Responder {
     let GithubQueryParams { owner, repo } = query.into_inner();
-    let apps = match LDNApplication::active(owner, repo, None).await {
-        Ok(app) => app,
-        Err(e) => return HttpResponse::BadRequest().body(e.to_string()),
-    };
-    HttpResponse::Ok().body(serde_json::to_string_pretty(&apps).unwrap())
+    match LDNApplication::active(owner, repo, None).await {
+        Ok(app) => match serde_json::to_string_pretty(&app) {
+            Ok(response) => HttpResponse::Ok().body(response),
+            Err(_) => HttpResponse::InternalServerError().body("Failed to parse success message"),
+        },
+        Err(e) => HttpResponse::BadRequest().body(e.to_string()),
+    }
 }
 
 #[get("/application/merged")]
 pub async fn merged(query: web::Query<GithubQueryParams>) -> actix_web::Result<impl Responder> {
     let GithubQueryParams { owner, repo } = query.into_inner();
     match LDNApplication::merged(owner, repo).await {
-        Ok(apps) => Ok(HttpResponse::Ok().body(serde_json::to_string_pretty(&apps).unwrap())),
+        Ok(apps) => match serde_json::to_string_pretty(&apps) {
+            Ok(response) => Ok(HttpResponse::Ok().body(response)),
+            Err(_) => {
+                Ok(HttpResponse::InternalServerError().body("Failed to parse success message"))
+            }
+        },
         Err(e) => Ok(HttpResponse::InternalServerError().body(e.to_string())),
     }
 }
 
 #[post("/application/notify_refill")]
 pub async fn notify_refill(info: web::Json<NotifyRefillInfo>) -> impl Responder {
     match LDNApplication::notify_refill(info.into_inner()).await {
-        Ok(()) => HttpResponse::Ok().body(serde_json::to_string_pretty("Success").unwrap()),
+        Ok(_) => match serde_json::to_string_pretty("Success") {
```

Comment:
to_string_pretty won't fail if passed a static string. It's a good candidate for .expect.